src/Sentry/SentryClient.cs

```diff
@@ -156,6 +156,9 @@ public void CaptureTransaction(SentryTransaction transaction, Scope? scope, Sent
             }
         }
 
+        // Set the user id to our fallback, if this hasn't already been set
+        transaction.User.Id ??= _options.InstallationId;
+
```

Comment:
We could. The Enricher appears to be applied here (for events):
https://github.com/getsentry/sentry-dotnet/blob/777d6649653b3361fd21b1c547316232442474bd/src/Sentry/Internal/MainSentryEventProcessor.cs#L157

And here (for transactions):
https://github.com/getsentry/sentry-dotnet/blob/37eb4a54cee682bec0d63435ce9959a67fb4656e/src/Sentry/SentryClient.cs#L145

This, in itself, is a bit odd. Why use the `Sentry.Client._enricher` to enrich transactions but use the `MainEventProcessor._enricher` to enrich events? It may be something to do with the order in which things get processed. In ASP.NET Core applications, for example, there is [logic in the ScopeExtensions](https://github.com/getsentry/sentry-dotnet/blob/1447b5598a6c5b23bffc7e3bb19deadb98088acc/src/Sentry.AspNetCore/ScopeExtensions.cs#L39-L51) (for Sentry instrumented applications) and in an [AspNetCoreEnricher](https://github.com/getsentry/sentry-dotnet/blob/bd62cf0421cc7a5d7b7804f8461fae00d47d43b8/src/Sentry.OpenTelemetry/AspNetCoreEnricher.cs) (for OpenTelemetry instrumented applications) that will only be applied if none of the properties for the `User` have been set. We would want that logic to be applied before setting the `User.Id` fallback.

In any event, we could move the logic from `SentryClient` to the `Enricher`. A couple of implications of doing this:
- The logic would then be applied before any transaction processors (that's when the enricher is applied for transactions). 
- We can't guarantee whether the logic gets applied before or after any custom event processors (the `MainSentryEventProcessor` that enriches events is just one of the event processors our SDK applies in a loop).

TLDR; I think it would make the code tidier but the behaviour less certain.

Do we still want to move it?